### ISSUE TITLE
Readme bug

### ISSUE THREAD

**Issue by schnittstabil** [equip/dispatch#6]
The Readme states:
```php
$stack = new Stack($middlewares, $default);
$response = $stack->dispatch($request);
```

However, currently that does not work. Actually, it have to state:
```php
$stack = new Stack($middlewares);
$response = $stack->dispatch($request, $default);
```

PRs which solves that:

* #1 – Fix Stack implementation according to README.md
* #2 – Fix README.md according to Stack implementation

It's more a philosophical question, which one to choose:

* #1 leads to #5 – and that Stacks can be seen as delegates too
* #2 leads to #4 – and that Stacks are middlewares too

**Comment by schnittstabil** [comment]
@shadowhand Done, please start reading here :wink:

**Comment by shadowhand** [comment]
Fixed by #4 and further addressed in #7.
